homeassistant/components/manual/alarm_control_panel.py

```diff
@@ -284,61 +284,61 @@ def code_format(self) -> alarm.CodeFormat | None:
             return alarm.CodeFormat.NUMBER
         return alarm.CodeFormat.TEXT
 
-    def alarm_disarm(self, code: str | None = None) -> None:
+    async def async_alarm_disarm(self, code: str | None = None) -> None:
         """Send disarm command."""
-        if not self._validate_code(code, STATE_ALARM_DISARMED):
+        if not self._async_validate_code(code, STATE_ALARM_DISARMED):
             return
 
         self._state = STATE_ALARM_DISARMED
         self._state_ts = dt_util.utcnow()
-        self.schedule_update_ha_state()
+        self.async_schedule_update_ha_state()
```

Comment:
`async_schedule_update_ha_state` is legacy. Please use `async_write_ha_state` instead.

Reply:
https://github.com/home-assistant/core/pull/84395